jupyterlite_sphinx/jupyterlite_sphinx.py

```diff
@@ -990,15 +989,38 @@ def jupyterlite_build(app: Sphinx, error):
             jupyterlite_contents = [jupyterlite_contents]
 
         # Expand globs in the contents strings
-        jupyterlite_contents = [
-            match
-            for pattern in jupyterlite_contents
-            for match in glob.glob(pattern, recursive=True)
-        ]
-
         contents = []
-        for content in jupyterlite_contents:
-            contents.extend(["--contents", content])
+        for pattern in jupyterlite_contents:
+            pattern_path = Path(pattern)
+
+            if pattern_path.is_absolute():
+                base_path = pattern_path.parent
+                glob_pattern = pattern_path.name
+            else:
+                # For relative paths, we'll resolve them relative
+                # to app.srcdir
+                base_path = Path(app.srcdir) / pattern_path.parent
+                glob_pattern = pattern_path.name
```

Comment:
```suggestion
            base_path = pattern_path.parent if pattern_path.is_abolute() else Path(app.srcdir) / pattern_path.parent
            glob_pattern = pattern_path.name
```

`glob_pattern` is being set to the same thing regardless of which branch is taken, so it can be moved out of the `if-else` block.